### PR TITLE
[DOCU-2187] Add config.canary_by_header_name to canary plugin

### PR DESCRIPTION
**Pull request by brentos**
### Summary
Add documentation for new config.canary_by_header_name for canary plugin

### Reason
Addresses DOCU-2187

### CODE REVIEW

File: app/_hub/kong-inc/canary/index.md [hunk 1]
```diff
@@ -14,6 +14,7 @@ categories:
 kong_version_compatibility:
   enterprise_edition:
     compatible:
+      - 2.8.x
       - 2.7.x
       - 2.6.x
       - 2.5.x
```
Comment (lena-larionova) [inline app/_hub/kong-inc/canary/index.md:20]: Should remove the previous versions, if version 0.6.x comes with gateway 2.8.x.

File: app/_hub/kong-inc/canary/index.md [hunk 2]
```diff
@@ -115,6 +116,13 @@ params:
       datatype: string
       description: |
         Header name whose value will be used as hash input. Required if `config.hash` is set to `header`.
+    - name: canary_by_header_name
+      required: null
+      default: null
+      value_in_examples: null
+      datatype: string
+      description: |
+        Header name that, when present on a request, will override the configured canary functionality. If the configured header is present with the value `always` then the request will always go to the canary upstream. If the header is present with the value `never` then the request will never go to the canary upstream. For all other values, the configured canary rules will be applied.
```
Comment (lena-larionova) [inline app/_hub/kong-inc/canary/index.md:125]:
```suggestion
        Header name that, when present on a request, overrides the configured canary 
        functionality. 
        
        * If the configured header is present with the value `always`, the request will 
        always go to the canary upstream. 
        * If the header is present with the value `never`, the request will never go to the 
        canary upstream. 
        
        For all other values, the configured canary rules will be applied.
```

File: app/_hub/kong-inc/canary/index.md [hunk 3]
```diff
@@ -167,6 +175,13 @@ _Service A_ and _Service B_ on consecutive requests.
 Canary provides an automatic fallback if, for some reason, a Consumer, Header, or IP can
 not be identified. The fallback order is `consumer`->`ip`->`none`.
 
+### Overriding the Canary
+
+In some cases, it may be desirable to allow clients to pick _Service A_ or _Service B_
```
Comment (lena-larionova) [inline app/_hub/kong-inc/canary/index.md:180]:
```suggestion
In some cases, you may want to allow clients to pick _Service A_ or _Service B_
```

**Review by lena-larionova — COMMENTED** [review]
Please also edit this file to add the new version: 

https://github.com/brentos/docs.konghq.com/blob/DOCU-2187-canary-override-header/app/_data/extensions/kong-inc/canary/versions.yml

As well, please add a changelog section to the bottom of the file:

```
---

## Changelog

### Kong Gateway 2.8.x (plugin version 0.6.0)

* Added the `canary_by_header_name` configuration parameter.
```

**Comment by brentos** [comment]
Thank you @lena-larionova! All proposed changes have been made

**Review by lena-larionova — APPROVED** [review]
LGTM! Thanks for your help!

**Pull request merged**